cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -87,6 +100,79 @@ public void run(FullHttpRequest request, HttpResponder responder) {
     }
   }
 
+  /**
+   * Run a task specified by query parameters: {@code className} and {@code param}
+   *
+   * @param request http request whose body contains {@link RunnableTaskRequest} in json format.
+   * @param responder {@link HttpResponder}
+   */
+
+  /**
+   * Run a task specified by query parameters {@code className} and {@code param}.
+   * <p>
+   * Request body contains the content of a file uploaded by the caller that is available to the task
+   * at runtime via {@link RunnableTaskContext}.
+   *
+   * @param request {@link HttpRequest} whose body should be a file content.
+   * @param responder {@link HttpResponder}
+   * @param className classname of the task to run
+   * @param param parameter(s) to the task
+   * @return {@link BodyConsumer}
+   */
+  @POST
+  @Path("/task")
+  public BodyConsumer runTask(HttpRequest request, HttpResponder responder,
+                              @QueryParam("className") String className,
+                              @QueryParam("param") String param) {
+    if (inflightRequests.incrementAndGet() > 1) {
+      responder.sendStatus(HttpResponseStatus.TOO_MANY_REQUESTS);
+      return null;
+    }
+
+    try {
+      // Download file content to local tmp dir.
+      final File dest = File.createTempFile("run_task_" + className, "tmp");
+
+      return new AbstractBodyConsumer(dest) {
+        @Override
+        protected void onFinish(HttpResponder responder, File uploadedFile) {
+          try {
+            RunnableTaskRequest runnableTaskRequest = new RunnableTaskRequest(className, param);
+            byte[] response = runnableTaskLauncher.launchRunnableTask(runnableTaskRequest, dest.toURI());
+            responder.sendByteArray(HttpResponseStatus.OK,
+                                    response,
```

Comment:
There is a chance that worker pod stops before finishing to send the response. This can be addressed in a followup PR.